[Bug]: 7.0.0 - Module '"@storybook/addon-actions/decorator"' has no exported member 'withActions'

### Describe the bug

According to [the 7.0 migration guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-auto-injection-of-storybookaddon-actions-decorator) I should be able to use the `withActions` decorator via `import { withActions } from '@storybook/addon-actions/decorator';`. But when I run TypeScript compile this gives me "error TS2305: Module '"@storybook/addon-actions/decorator"' has no exported member 'withActions'."

It looks like `withActions` [is exported from `addons/actions/src/decorator.ts`](https://github.com/storybookjs/storybook/blob/d9d2fbb31406c0783f412a43e6e37b7bd0e6684d/code/addons/actions/src/decorator.ts#L54)  but is [not exported from `addons/actions/src/index.ts`](https://github.com/storybookjs/storybook/blob/d9d2fbb31406c0783f412a43e6e37b7bd0e6684d/code/addons/actions/src/index.ts). This differs from other decorators like `withLinks` which [are exported from their index.ts](https://github.com/storybookjs/storybook/blob/d9d2fbb31406c0783f412a43e6e37b7bd0e6684d/code/addons/links/src/index.ts#L20).

It looks like this is intended to work via the ["exports" field in `package.json`](https://github.com/storybookjs/storybook/blob/c31eb0362eaf42bc889a9cca81a5dcf0767056da/code/addons/actions/package.json#L31) for declaring multiple entry points in `package.json`. That feature [gained support in TypeScript 4.7.0](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing). But I tried upgrading my project from TypeScript 4.6.4 to 4.7.0 and still saw the same error.

### To Reproduce

This is [reproducing](https://github.com/ni/nimble/actions/runs/4597590732) in [this branch of the ni/nimble repository](https://github.com/ni/nimble/blob/f1114dfd19a52f88f95d0149c2d42a8e422a280e/packages/nimble-components/src/button/tests/button.stories.ts#L3). To reproduce, clone that repo, checkout that branch, run `npm i` and `npm run build`. If a more targeted reproduction would be helpful I can try to create one.

### System

```shell
Environment Info:

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 16.15.1 - /usr/local/bin/node
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 111.0.5563.146
    Firefox: 111.0.1
    Safari: 16.3
```


### Additional context

This was also reported in https://github.com/storybookjs/storybook/issues/21308 but that bug is marked as fixed and I'm still seeing the problem. I don't have permission to re-open that issue and I was worried my comment there would get lost so I filed a new one; sorry if that's not the correct workflow!

--- a/code/addons/actions/package.json
+++ b/code/addons/actions/package.json
@@ -61,6 +61,9 @@
       "manager": [
         "dist/manager.d.ts"
       ],
+      "decorator": [
+        "dist/decorator.d.ts"
+      ],
       "preview": [
         "dist/preview.d.ts"
       ]

